docs/content/projects/internal/BuildMetrics.md

```diff
@@ -27,31 +27,25 @@ They will be referred in docs as `<placeholder>`.
 
 [Http build cache](https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_configure_remote) errors:
 
-- `<namespace>.build.cache.errors.[load|store].<http status code>`: errors counter
+- `<namespace>.<environment>.<node>.build.cache.errors.[load|store].<http status code>`: errors counter
```

Comment:
I want to see series name builder compatible with this concept
For example:
```
SeriesNameBuilder()
.addNamespace("namespace")
.addEnvironment(Environment.CI)
addNode() - I don't know what it is
.addProduct/Service("build.cache")
.addMetric("errors.load")
.addMetricState("200")
```

Reply:
I see the idea. I've considered this model also but decided to use only original stastd model. At least for now.

On this example:
- Namespace: it is on low level of statsd transport. Client can't override it in a metric.
- Environment, node (this is mostly a "user" for simplicity): want to see them as required. I tried couple of time to make them optional, but alway found that we need them in all metrics.
- Product/service VS metric: I can't describe the different between these terms. If I introduce a new term or abstraction, I need to understand cases, problems around it. For now, the difference is not clear and more confuses me.
- Metric state: this is metric type

So, it turns out a couple of options:

```
buildMetricSeriesNameBuilder   // with required defaults
	.addMetric("errors.load")    // series name
	.build() 				   // builds series name
```

Or an abstraction above stasd metric:

```
buildMetricBuilder	
	.addMetric("errors.load") 		// series name
	.timeMs() |gauge() | count()    // builds statsd metric
```

Reply:
Discussed internally, the idea:

```
@Service("build")
interface Build {
	@Path("cache")	
	val cache: Cache
	interface Cache {
		@Path("errors")
		val cache: Errors
		interface Errors {
			@GaugeMetric("load")
			fun load(value: Long)
		}
	}
}
build.cache.errors.load(100)
```